controllers/resources_test.go

```diff
@@ -183,4 +186,51 @@ var _ = Describe("ReconcileRolloutManager tests", func() {
 		}
 	})
 
+	It("Verify whether RolloutManager creating ServiceMonitor", func() {
+		smCRD := &crdv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "servicemonitors.monitoring.coreos.com",
+				Namespace: req.Namespace,
+			},
+		}
+
+		Expect(r.Client.Create(ctx, smCRD)).To(Succeed())
+
+		existingSvc := &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      DefaultArgoRolloutsMetricsServiceName,
+				Namespace: req.Namespace,
+				Labels: map[string]string{
+					"app.kubernetes.io/name":      DefaultArgoRolloutsMetricsServiceName,
+					"app.kubernetes.io/component": "server",
+				},
+			},
+			Spec: corev1.ServiceSpec{
+				Ports: []corev1.ServicePort{
+					{
+						Name:       "metrics",
+						Port:       8090,
+						Protocol:   corev1.ProtocolTCP,
+						TargetPort: intstr.FromInt(8090),
+					},
+				},
+				Selector: map[string]string{
+					DefaultRolloutsSelectorKey: DefaultArgoRolloutsResourceName,
+				},
+			},
+		}
+
+		Expect(r.Client.Create(ctx, existingSvc)).To(Succeed())
+
+		res, err := r.Reconcile(ctx, req)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res.Requeue).Should(BeFalse(), "reconcile should not requeue request")
+
+		sm := &monitoringv1.ServiceMonitor{}
+		Expect(r.Client.Get(ctx, types.NamespacedName{
+			Name:      DefaultArgoRolloutsMetricsServiceName,
+			Namespace: testNamespace,
+		}, sm)).To(Succeed())
+	})
+
 })
```

Comment:
Would also be beneficial to make sure ServiceMonitor is deleted if the RolloutManager no longer exists.